canonical-web-and-design/maas-ui
# feat(ui): change device routes to react version

## Done

- Replaced all instances of legacy device routes to react device routes
- Added redirects
- Driveby fix of "Add discovery" form taking the user to device details in instead of machine details when defining a parent

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Run the full app with `yarn start`
- Check you get taken to the React devices pages for the following:
  - Click on "Devices" in the header
  - Go to `/MAAS/r/domains`, click on a domain that has a resource record pointing at a device (e.g. "test" on bolla) then click on the device name
  - Go to `/MAAS/r/zones` and click on a devices count
  - Go to `/MAAS/r/settings/dhcp` and click on a DHCP snippet that applies to a device (e.g. "test-device" on bolla)
  - Go to `/MAAS/r/dashboard`, open the "Add discovery" form, then save the form with "Save and go to device listing"
- Check that redirects are in place
  - `/MAAS/l/devices` => `/MAAS/r/devices`
  - `/MAAS/l/device/abc123?area=network` => `/MAAS/r/device/abc123/network`

## Fixes

Fixes canonical-web-and-design/app-tribe#544
